### DISCUSSION TITLE
Best IDE for Monkey C?

### DISCUSSION THREAD

**Discussion by SandervanGeel** [matco/badminton#29] [Q&A]
Hi,

First of all nice application! I have started creating my own application for score keeping in different sport.
I am developing in visual studio code with the monkey c plugin.

The main struggles I have with this IDE is that auto completion is not working as I expect and the find reference / implementation does not seem to work:
![image](https://github.com/user-attachments/assets/faa0d6fb-49f6-4165-ab21-9a21643c41cf)
![image](https://github.com/user-attachments/assets/74f174a5-009b-4f87-9776-5e60fe0286a0)

auto completion:
![image](https://github.com/user-attachments/assets/b18209fc-c45c-4e6d-97e6-2c41734eab8e)

What IDE do you use and which plugins did you install?

**Comment by matco** [comment]
Hi!

Thanks for your message.  I'm also using Visual Studio Code with the Monkey C plugin and experiencing the same behavior you described.  I'm not surprised because monkey C is duck typed.

I spent considerable time setting up the types in my application, so I can use the "strict" mode of the compiler. I find it very valuable to catch errors early in the development process. But this is not mandatory, and I think many users are less concerned with type safety.
